src/xcode/ENA/ENA/Source/Services/PPAnalyticsSubmitter/PPAnalyticsSubmitter.swift

```diff
@@ -0,0 +1,321 @@
+////
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+import OpenCombine
+
+protocol PPAnalyticsSubmitting {
+	/// Triggers the submission of all collected analytics data. Only if all checks success, the submission is done. Otherwise, the submission is aborted. In all cases, NO completion is called.
+	func triggerSubmitData()
+
+	/// ONLY FOR TESTING. Triggers the submission of all collected analytics data. Only if all checks success, the submission is done. Otherwise, the submission is aborted. The completion calls are passed through to test the component.
+	func triggerSubmitData(ppacToken: PPACToken?, completion: ((Result<Void, PPASError>) -> Void)?)
+}
+
+final class PPAnalyticsSubmitter: PPAnalyticsSubmitting {
+
+	// MARK: - Init
+
+	init(
+		store: Store,
+		client: Client,
+		appConfig: AppConfigurationProviding
+	) {
+		self.store = store
+		self.client = client
+		self.configurationProvider = appConfig
+	}
+
+	// MARK: - Protocol PPAnalyticsSubmitting
+
+	func triggerSubmitData() {
+		triggerSubmitData(ppacToken: nil, completion: nil)
```

Comment:
does that make sense to submit without a token? Or will this changed with an other PR?

Reply:
This is just for testing purposes. So when the token is nil, a token is requested to create and authorize. For unit tests, this will fail (Simulator).
So for me it is okay as it is.